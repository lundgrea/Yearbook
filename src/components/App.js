import React, { Component } from 'react';
import Cohort from './Cohort';
import people from '../data/yearbook-data.js';
import PersonForm from './PersonForm';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: people.staff,
      students: people.students
    }
  }

  addNewPerson = (newPerson) => {
    this.setState({students: [...this.state.students, newPerson]})
  }

  deletePerson = (id) => {
    console.log('inside the delete method')
    const filteredStaff = this.state.staff.filter(person => person.id !== id)
    this.setState({ staff: filteredStaff})
    const filteredStudents = this.state.students.filter(student => student.id !== id)
    this.setState({ students: filteredStudents})
  }

  render() {
    return (
      <main className="App">
        <header className="App-header">
          <h1>Turing Yearbook</h1>
        </header>
        <PersonForm addNewPerson={this.addNewPerson}/>
        <Cohort title="Staff" data={this.state.staff} deletePerson={this.deletePerson} />
        <Cohort title="Students" data={this.state.students} deletePerson={this.deletePerson}/>
      </main>
    );
  }
}

export default App;
