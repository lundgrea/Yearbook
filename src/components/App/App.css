html {
  height: 100%;
  overflow: hidden;
}

body {
  background-color:#026475;;
  background-size: cover;
  background-attachment: fixed;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  overflow: auto;
}

.App {
  text-align: center;
}

h1 {
  padding: 10px;
  color: #51CFDE;
  font-family: helvetica neue heavy;
  font-size: 72px;
  font-weight: 900;
  letter-spacing: -4px;
}

.App-header{
  display: flex
}

.App-header-logo{
  margin: 50px 5px 50px 50px;
}