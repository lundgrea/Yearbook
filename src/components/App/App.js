import React, { Component } from 'react';
import Cohort from '../Cohort/Cohort';
import people from '../../data/yearbook-data.js';
import PersonForm from '../PersonForm/PersonForm';
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      staff: people.staff,
      students: people.students
    }
  }

  addNewPerson = (newPerson) => {
    this.setState({students: [...this.state.students, newPerson]})
  }

  deletePerson = (id) => {
    const filteredStaff = this.state.staff.filter(person => person.id !== id)
    this.setState({ staff: filteredStaff})
    const filteredStudents = this.state.students.filter(student => student.id !== id)
    this.setState({ students: filteredStudents})
  }

  updatePerson = (event) => {
    console.log('in the update method')
    this.setState({[event.target.name]: event.target.value})
  }

  render() {
    return (
      <main className="App">
        <header className="App-header">
          <img className="App-header-logo" alt="Turing Logo" height="100px" width="100px" src="https://d682ma8ami8n4.cloudfront.net/images/TuringSchool_LogoMark_Gray.png"/>
          <h1>TURING YEARBOOK</h1>
        <PersonForm className="form" addNewPerson={this.addNewPerson}/>
        </header>
        <Cohort title="Staff" data={this.state.staff} deletePerson={this.deletePerson} updatePerson={this.updatePerson} />
        <Cohort title="Students" data={this.state.students} deletePerson={this.deletePerson} updatePerson={this.updatePerson}/>
      </main>
    );
  }
}

export default App;
