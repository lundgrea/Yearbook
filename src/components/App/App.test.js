import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    const wrapper = shallow(<App
    />)
  expect(wrapper).toMatchSnapshot()
  })

  it('should update state with a new person when addNewPerson is called', () => {
    // Setup
    const wrapper = shallow(<App />);
    const mockPerson = { 
      id: 1, 
      name: 'Eric', 
      quote: 'You miss all the shots you don\'t take.', 
      superlative: 'Most Likely to Win the PGA', 
      photo: './eric.jpg'
     };
     const expected = [mockPerson]

    // Expectation
    // expect(wrapper.state('staff')).toEqual([]);

    // Execution
    wrapper.instance().addNewPerson(mockPerson);

    // Expectation
    expect(wrapper.state('staff')).toHaveLength(13)
  })
})