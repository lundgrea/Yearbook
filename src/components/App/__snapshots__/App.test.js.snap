// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot with all data passed in correctly 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <h1>
      Turing Yearbook
    </h1>
  </header>
  <PersonForm
    addNewPerson={[Function]}
  />
  <Cohort
    data={
      Array [
        Object {
          "id": 1,
          "name": "Eric",
          "photo": "./eric.jpg",
          "quote": "You miss all the shots you don't take.",
          "superlative": "Most Likely to Win the PGA",
        },
        Object {
          "id": 2,
          "name": "Amy",
          "photo": "./amy.jpg",
          "quote": "I believe in you, cuties!",
          "superlative": "Most Likely to Be an Actual Superhero",
        },
        Object {
          "id": 3,
          "name": "Khalid",
          "photo": "./khalid.png",
          "quote": "Don't put the world on your shoulders.",
          "superlative": "Most Likely to Save the Planet from Alien Invasion",
        },
        Object {
          "id": 4,
          "name": "Travis",
          "photo": "./travis.jpg",
          "quote": "You do you.",
          "superlative": "Most Likely to Win a Grammy",
        },
        Object {
          "id": 5,
          "name": "Leta",
          "photo": "./leta.jpg",
          "quote": "It's okay to be kind of a mess.",
          "superlative": "Most Likely to Go to the Moon",
        },
        Object {
          "id": 6,
          "name": "Christie",
          "photo": "./christie.jpg",
          "quote": "Every journey begins with a single step. So take the damn step!",
          "superlative": "Most Likely to Be President",
        },
        Object {
          "id": 7,
          "name": "Brittany",
          "photo": "./brittany.jpg",
          "quote": "It was Robbie.",
          "superlative": "Most Likely to Get Away with Murder",
        },
        Object {
          "id": 8,
          "name": "Robbie",
          "photo": "./robbie.jpg",
          "quote": "Cool.",
          "superlative": "Most Likely to Be Accused of Murder",
        },
        Object {
          "id": 9,
          "name": "Will",
          "photo": "./will.jpg",
          "quote": "Stop worrying and just do the thing.",
          "superlative": "Most Likely to Invent Cold Fusion",
        },
        Object {
          "id": 10,
          "name": "Pam",
          "photo": "./pam.jpg",
          "quote": "Shit!",
          "superlative": "Most Likely to Win a Gold Medal",
        },
        Object {
          "id": 11,
          "name": "David",
          "photo": "./david.JPG",
          "quote": "I love git.",
          "superlative": "Most Likely to Be Eaten by a Shark",
        },
        Object {
          "id": 12,
          "name": "Louisa",
          "photo": "./louisa.jpg",
          "quote": "YOLO",
          "superlative": "Most Likely to Win a Knife Fight",
        },
        Object {
          "id": 13,
          "name": "Jeff",
          "photo": "https://i.ytimg.com/vi/DhqzMc_LXgQ/maxresdefault.jpg",
          "quote": "::thoughtful nonsense::",
          "superlative": "Most Likely to Tell a Dad Joke",
        },
      ]
    }
    deletePerson={[Function]}
    title="Staff"
    updatePerson={[Function]}
  />
  <Cohort
    data={
      Array [
        Object {
          "id": 21,
          "name": "Dr. Suess",
          "photo": "https://placekitten.com/200/300",
          "quote": "Don't cry because it's over, smile because it happened",
          "superlative": "",
        },
        Object {
          "id": 22,
          "name": "Oscar Wilde",
          "photo": "https://placekitten.com/200/300",
          "quote": "Be yourself; everyone else is already taken",
          "superlative": "",
        },
        Object {
          "id": 23,
          "name": "Frank Zappa",
          "photo": "https://placekitten.com/200/300",
          "quote": "So many books, so little time.",
          "superlative": "",
        },
        Object {
          "id": 24,
          "name": "Danesh",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 25,
          "name": "Evelyn",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 26,
          "name": "Fatima",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 27,
          "name": "Glen",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 28,
          "name": "Harriet",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 29,
          "name": "Isabella",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 210,
          "name": "Jorge",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 211,
          "name": "Kelly",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 212,
          "name": "Lucia",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 213,
          "name": "Mei",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 214,
          "name": "Nina",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 215,
          "name": "Ophus",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 216,
          "name": "Parker",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 217,
          "name": "Quin",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 218,
          "name": "Rosa",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 219,
          "name": "Sam",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 220,
          "name": "Tyrese",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 221,
          "name": "Ursula",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 222,
          "name": "Vic",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 223,
          "name": "Whitney",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 224,
          "name": "Xavier",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 225,
          "name": "Yolanda",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
        Object {
          "id": 226,
          "name": "Zach",
          "photo": "https://placekitten.com/200/300",
          "quote": "",
          "superlative": "",
        },
      ]
    }
    deletePerson={[Function]}
    title="Students"
    updatePerson={[Function]}
  />
</main>
`;
